src/test_discovery.ts

```diff
@@ -77,6 +79,8 @@ export async function refresh_tests(
           test_item.tags = get_test_tags(test.properties);
           test_item.description = get_test_description(test.properties);
           tests.push(test_item);
+
+          test_details.set(test_item, { command: test.command });
```

Comment:
I think you need to clear out the `test_details` before entering this loop. It's possible a user will remove or rename tests. When that happens, the data for the old test name will clutter the `test_details`.

Reply:
As a week_map, test_details can only be accessed by the keys it stores (weak_map is not iterable). When those keys are garbage collected, they are removed from test_details. Since we are replacing the list of tests in the controller, I think that takes care of the stale test data. test_details should take care of itself.